src/SFML/Audio/SoundFileWriterOpus.cpp

```diff
@@ -0,0 +1,259 @@
+    }
+    int status = 0;
+    m_opus = opus_encoder_create(sampleRate, channelCount, OPUS_APPLICATION_AUDIO, &status);
+    if (status != OPUS_OK)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\"" << std::endl;
+        if (status == OPUS_BAD_ARG)
+            err() << "Possible wrong sample rate, allowed are 8000, 12000, 16000, 24000, or 48000 Hz." << std::endl;
+        close();
+        return false;
+    }
+
+    m_file.open(filename.c_str(), std::ios::binary);
+    if (!m_file)
+    {
+        err() << "Failed to write ogg/opus file \"" << filename << "\" (cannot open file)" << std::endl;
+        close();
+        return false;
+    }
+
+    // Set bitrate (VBR is default)
+    opus_encoder_ctl(m_opus, OPUS_SET_BITRATE(128000));
+
+    // Create opus header
+    unsigned char headerData[19];
+    memset(static_cast<void*>(headerData), 0, 19);
+    memcpy(static_cast<void*>(headerData), "OpusHead", 8);
+    headerData[8] = 1; // Version
+    headerData[9] = channelCount;
+    writeUint32(headerData + 12, static_cast<Uint32>(sampleRate));
+    headerData[18] = channelCount > 8 ? 255 : (channelCount > 2); // Mapping family
+
+    // Map opus header to ogg packet
+    ogg_packet op;
+    op.packet = headerData;
+    op.bytes = 19;
+    op.b_o_s = 1;
+    op.e_o_s = 0;
+    op.granulepos = 0;
+    op.packetno = m_packageNumber++;
+
+    // Write the header packet to the ogg stream
+    ogg_stream_packetin(&m_ogg, &op);
+    flushBlocks();
+
+    // Create comment header, needs to be in a new page
+    const char* opusVersion = opus_get_version_string();
+    Uint32 opusVersionLength = strlen(opusVersion);
+    const int commentLength  = 8 + 4 + opusVersionLength + 4;
+
+    unsigned char commentData[commentLength];
+
+    // Magic bytes
+    memcpy(static_cast<void*>(commentData), "OpusTags", 8);
+
+    // unsigned 32bit integer: Length of vendor string (encoding library)
+    writeUint32(commentData + 8, opusVersionLength);
+
+    // Vendor string
+    memcpy(static_cast<void*>(commentData+12), opusVersion, opusVersionLength);
+
+    // Length of user comments (E.g. you can add a ENCODER tag for SFML)
+    writeUint32(commentData + 12+opusVersionLength, 0);
+
+    op.packet     = commentData;
+    op.bytes      = commentLength;
+    op.b_o_s      = 0;
+    op.e_o_s      = 0;
+    op.granulepos = 0;
+    op.packetno   = m_packageNumber++;
+    ogg_stream_packetin(&m_ogg, &op);
+
+    // This ensures the actual audio data will start on a new page, as per spec
+    flushBlocks();
+
+    return true;
+}
+
+
+////////////////////////////////////////////////////////////
+void SoundFileWriterOpus::write(const Int16* samples, Uint64 count)
+{
+    assert(m_opus);
+
+    const opus_uint32 frame_size = 960;
+    const opus_int32 buffer_size = frame_size * m_channelCount * sizeof(Int16);
+    unsigned char buffer[buffer_size];
+    Uint32 frame_number = 0;
+    Uint8 endOfStream = 0;
+    ogg_packet op;
```

Comment:
I would declare the variables that are not reused between iterations (such as `op`) locally inside the loop. It makes code easier to understand.